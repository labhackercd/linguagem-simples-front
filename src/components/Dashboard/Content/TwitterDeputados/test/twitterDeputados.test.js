import React from 'react';
import { render, screen } from '@testing-library/react';
import {shallow} from "enzyme/build";
import TwitterDeputadosContent from '../twitterDeputados'
import { mount} from 'enzyme';

it("should render the TwitterDeputados section", () => {
    const component = shallow(<TwitterDeputadosContent/>);

    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

test("Test agencia content lifeclycle with search", async (done) => {

            
    const wrapper = await mount(<TwitterDeputadosContent sessionId={1}/>);


        wrapper.update();
        //console.log(wrapper.debug())

        const searchField = wrapper.find("input").at(0);
        //console.log(searchField.debug());
        searchField.instance().value = "presential";
        searchField.simulate("change");
        //expect(wrapper.find("input").at(0).prop('value')).toEqual("presential");
        

        done();


  });