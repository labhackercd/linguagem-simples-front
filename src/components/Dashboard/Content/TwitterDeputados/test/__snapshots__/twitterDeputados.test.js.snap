// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the TwitterDeputados section 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={8}
    >
      <WithStyles(ForwardRef(Typography))
        style={
          Object {
            "color": "#007E5A",
          }
        }
        variant="h6"
      >
        Mais recentes 
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        InputProps={
          Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="start"
            >
              <UNDEFINED />
            </ForwardRef(WithStyles)>,
          }
        }
        id="input-search-agencia"
        onChange={[Function]}
        size="small"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <Styled(MuiBox)
        paddingTop={3}
      >
        <List
          direction="ltr"
          height={250}
          itemCount={515}
          itemSize={44}
          layout="vertical"
          overscanCount={2}
          useIsScrolling={false}
        >
          <Component />
        </List>
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
